demo/src/utils.js

```diff
@@ -0,0 +1,127 @@
+      );
+      const loader = await createTiffPyramid({ channelUrls });
+      return loader;
+    }
+    case 'static': {
+      const loader = await createZarrLoader(infoObj);
+      return loader;
+    }
+    default:
+      throw Error(`Pyramid type (${type}) is not supported`);
+  }
+}
+
+export function useWindowSize(scaleWidth = 1, scaleHeight = 1) {
+  function getSize() {
+    return {
+      width: window.innerWidth * scaleWidth,
+      height: window.innerHeight * scaleHeight
+    };
+  }
+
+  const [windowSize, setWindowSize] = useState(getSize());
+
+  useEffect(() => {
+    const handleResize = () => {
+      setWindowSize(getSize());
+    };
+    window.addEventListener('resize', handleResize);
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  });
+  return windowSize;
+}
+
+export function channelsReducer(state, { index, value, type }) {
+  switch (type) {
+    case 'CHANGE_CHANNEL': {
+      const { name, selection } = value;
+      const names = [...state.names];
+      const selections = [...state.selections];
+      names[index] = name;
+      selections[index] = selection;
+      return { ...state, names, selections };
+    }
+    case 'CHANGE_COLOR': {
+      const colors = [...state.colors];
+      colors[index] = value;
+      return { ...state, colors };
+    }
+    case 'CHANGE_SLIDER': {
+      const sliders = [...state.sliders];
+      sliders[index] = value;
+      return { ...state, sliders };
+    }
+    case 'TOGGLE_ON': {
+      const isOn = [...state.isOn];
+      isOn[index] = !isOn[index];
+      return { ...state, isOn };
+    }
+    case 'ADD_CHANNEL': {
+      const { name, selection, color } = value;
+      const nextState = { ...state };
+      nextState.names.push(name);
+      nextState.selections.push(selection);
+      nextState.colors.push(color);
+      nextState.isOn.push(true);
+      nextState.sliders.push([0, 20000]);
+      nextState.ids.push(String(Math.random()));
+      return nextState;
+    }
+    case 'REMOVE_CHANNEL': {
+      const names = state.names.filter((_, i) => i !== index);
+      const sliders = state.sliders.filter((_, i) => i !== index);
+      const colors = state.colors.filter((_, i) => i !== index);
+      const isOn = state.isOn.filter((_, i) => i !== index);
+      const ids = state.ids.filter((_, i) => i !== index);
+      const selections = state.selections.filter((_, i) => i !== index);
+      return { names, sliders, colors, isOn, ids, selections };
+    }
+    case 'RESET_CHANNELS': {
+      const { names, selections = [] } = value;
+      const n = names.length;
+      return {
+        names,
+        selections,
+        sliders: Array(n).fill([0, 20000]),
+        colors: range(n).map(i => DEFAULT_COLOR_PALLETE[i]),
+        isOn: Array(n).fill(true),
+        ids: range(n).map(() => String(Math.random()))
+      };
+    }
+    default:
+      throw new Error();
+  }
+}
+
+export function hexToRgb(hex) {
```

Comment:
Nothing for you to do but we should probably talk to Nils about #104 sooner rather than later.